src/fromager/resolver.py

```diff
@@ -145,14 +146,27 @@ def identify(self, requirement_or_candidate: Requirement | Candidate) -> str:
     def get_extras_for(
         self,
         requirement_or_candidate: Requirement | Candidate,
-    ) -> tuple[str]:
+    ) -> tuple[str, ...]:
         # Extras is a set, which is not hashable
-        return tuple(sorted(requirement_or_candidate.extras))
+        if requirement_or_candidate.extras:
+            return tuple(sorted(list(requirement_or_candidate.extras)))
```

Comment:
`sorted()` works with any iterable

```suggestion
            return tuple(sorted(requirement_or_candidate.extras))
```

Reply:
the `Requirement` type defines `extras` as a set which needs to be converted to a list